lib/ts/fetch.ts

```diff
@@ -685,5 +698,13 @@ export async function setFrontToken(frontToken: string | undefined) {
         }
     }
 
+    const oldToken = await getFrontToken();
+    if (oldToken !== null && frontToken !== undefined) {
+        const oldPayload = parseFrontToken(oldToken).up;
+        const newPayload = parseFrontToken(frontToken).up;
+        if (JSON.stringify(oldPayload) !== JSON.stringify(newPayload)) {
+            onTokenUpdate();
+        }
+    }
```

Comment:
This may get triggered during session refresh as well - if the refresh API causes a change in the access token payload. I think it should fire even during session refresh - while from auth-react's point of view this will cause a double update, semantically, firing the `ACCESS_TOKEN_PAYLOAD_UPDATED ` event still makes sense.

What do you think?

Reply:
We should also add a test for this where a session refresh causes a change in the access token payload and then on session refresh, we check that `ACCESS_TOKEN_PAYLOAD_UPDATED` is fired.

Reply:
Yep, I think that makes sense. It already works that way, so I've added a test for it.